etc/sync-function-template.js

```diff
@@ -339,13 +351,10 @@ function(doc, oldDoc) {
       itemProperties.push(itemProp);
     }
 
-    var oldItemPropertiesCount = 0;
     for (var oldItemProp in oldItemValue) {
-      oldItemPropertiesCount++;
-    }
-
-    if (itemProperties.length !== oldItemPropertiesCount) {
-      return false;
+      if (itemProperties.indexOf(oldItemProp) < 0) {
+        itemProperties.push(oldItemProp);
+      }
     }
```

Comment:
Null and undefined are now treated as effectively equal when determining if a document or object property has been modified. Previously they were not because the old and new objects' property counts would differ if a property was null in one and undefined in the other. Now the property counts don't affect the test.